akka-http/src/main/scala/akka/http/scaladsl/server/Directive.scala

```diff
@@ -45,9 +46,10 @@ abstract class Directive[L](implicit val ev: Tuple[L]) {
     def validatedMap[R](f: L => R)(implicit tupler: Tupler[R]): Directive[tupler.Out] =
       Directive[tupler.Out] { inner =>
         tapply { values => ctx =>
-          try inner(tupler(f(values)))(ctx)
-          catch {
-            case e: IllegalArgumentException => ctx.reject(ValidationRejection(e.getMessage.nullAsEmpty, Some(e)))
+          Try(f(values)) match {
```

Comment:
perhaps try/catch is a little more efficient since it saves some allocations?

Reply:
I tried first try/catch way. But I couldn't work out code with try/catch because try/catch have not "not throwed case" (like ruby's else of "begin - rescue - else").

If you have good idea, I'm very happy :)

Reply:
Hmm, you can break this out to a separate function and use 'return' in the 'catch' block... but I agree that's a bit weird perhaps this is OK then :)

Reply:
```scala
try {
    val r = f(values)
    inner(tupler(r))(ctx) // ☆
} catch {
    case e: IllegalArgumentException => ctx.reject(ValidationRejection(e.getMessage.nullAsEmpty, Some(e)))
}
```

This code doesn't solve the problem because ☆ code is stiil inside of try/catch.

```scala
val rOrRejected = try {
    f(values)
} catch {
    case e: IllegalArgumentException => ctx.reject(ValidationRejection(e.getMessage.nullAsEmpty, Some(e))) // △
}

inner(tupler(r))(ctx) // XD  r is isntace of R or Future[RouteResult], result of △
```

But if I put ☆ outside of try-catch, I have to handle concrete r case and Future[RouteResult].

Reply:
> Hmm, you can break this out to a separate function and use 'return' in the 'catch' block... but I agree that's a bit weird perhaps this is OK then :)

Great Idea! Wait a moment ~